### PR TITLE
build: macOS fix Info.plist

### PR DESCRIPTION
**Pull request by RandyMcMillan**
This change updates the Info.plist that is bundled with the macOS app.

It removes references to bitcoinfoundation.org
It updates the copyright years -- section 2205.1(F)
It capitalizes "Developers" -- section 2204.1

This PR is related to #17311
sections -- 2207.1(E), 2207.2, 2207.9, 2206.1

REF: https://www.copyright.gov/comp3/chap2200/ch2200-notice.pdf

***old***
Screen Shot 2019-11-12 at 9 49 28 PM
***new***
Screen Shot 2019-11-12 at 9 49 08 PM

### CODE REVIEW

File: share/qt/Info.plist.in [hunk 1]
```diff
@@ -38,15 +38,15 @@
   <true/>
 
   <key>CFBundleIdentifier</key>
-  <string>org.bitcoinfoundation.Bitcoin-Qt</string>
+  <string>org.bitcoincore.Bitcoin-Qt</string>
```
Comment (jonasschnelli) [inline share/qt/Info.plist.in]: I think this could have further consequences... though not sure.
* codesigning?
* internal paths for preferences/settings?

Comment (emilengler) [reply share/qt/Info.plist.in]: This change is dangerous. It will wipe out the settings
https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleidentifier

**Comment by MarcoFalke** [comment]
NACK, as this will wipe all user settings.

**Comment by RandyMcMillan** [comment]
We could append the major and minor release to the app name. Example “Bitcoin Core-0.2.0.app” 

This would preserve the existing installed app but allow for upgrades moving forward.

File: share/qt/Info.plist.in [hunk 2]
```diff
@@ -17,10 +17,10 @@
   <string>APPL</string>
 
   <key>CFBundleGetInfoString</key>
-  <string>@CLIENT_VERSION_MAJOR@.@CLIENT_VERSION_MINOR@.@CLIENT_VERSION_REVISION@.@CLIENT_VERSION_BUILD@, Copyright © 2009-@COPYRIGHT_YEAR@ @COPYRIGHT_HOLDERS_FINAL@</string>
+  <string>Copyright © 2009-@COPYRIGHT_YEAR@ @COPYRIGHT_HOLDERS_FINAL@</string>
 
   <key>CFBundleShortVersionString</key>
-  <string>@CLIENT_VERSION_MAJOR@.@CLIENT_VERSION_MINOR@.@CLIENT_VERSION_REVISION@</string>
```
Comment (fanquake) [inline share/qt/Info.plist.in]: Why are you removing @CLIENT_VERSION_REVISION?

File: Makefile.am [hunk 3]
```diff
@@ -321,4 +321,3 @@ clean-local: clean-docs
 	rm -rf coverage_percent.txt test_bitcoin.coverage/ total.coverage/ test/tmp/ cache/ $(OSX_APP)
 	rm -rf test/functional/__pycache__ test/functional/test_framework/__pycache__ test/cache share/rpcauth/__pycache__
 	rm -rf osx_volname dist/ dpi36.background.tiff dpi72.background.tiff
-
```
Comment (fanquake) [inline Makefile.am]: Please don't make unrelated changes in PRs.

**Review by fanquake — COMMENTED** [review]
> We could append the major-minor-client release to the app name. Example “Bitcoin Core-0.2.0.app”

NACK. Also NACK the `CFBundleIdentifier` identifier changes.

File: share/qt/Info.plist.in (hunk 2 above)
Comment (RandyMcMillan) [reply share/qt/Info.plist.in]: That is the SHORT version string. This should be used for major releases (in this case we should include the minor release number but the client number is for the regular version string. 


I will update the PR and add better explanations soon.

Comment (RandyMcMillan) [reply share/qt/Info.plist.in]: One of the goals in this PR will result in better issue tracking for the macOS builds.

File: share/qt/Info.plist.in (hunk 1 above)
Comment (RandyMcMillan) [reply share/qt/Info.plist.in]: Yes - there are "consequences" to the changes proposed in this PR. The long term benefits IMO out weigh the short term transition with the goals of this PR. More explanations soon.

File: configure.ac [hunk 4]
```diff
@@ -7,7 +7,7 @@ define(_CLIENT_VERSION_BUILD, 0)
 define(_CLIENT_VERSION_RC, 0)
 define(_CLIENT_VERSION_IS_RELEASE, false)
 define(_COPYRIGHT_YEAR, 2019)
-define(_COPYRIGHT_HOLDERS,[The %s developers])
+define(_COPYRIGHT_HOLDERS,[The %s Developers])
```
Comment (laanwj) [inline configure.ac]: Please don't change this.

File: share/qt/Info.plist.in (hunk 2 above)
Comment (fanquake) [reply share/qt/Info.plist.in]: > That is the SHORT version string. 

I'd rather use the full version string, to avoid any ambiguity. Shortened version strings have caused issues in the past. See #4029 and #14697.

File: share/qt/Info.plist.in [hunk 5]
```diff
@@ -17,10 +17,10 @@
   <string>APPL</string>
 
   <key>CFBundleGetInfoString</key>
-  <string>@CLIENT_VERSION_MAJOR@.@CLIENT_VERSION_MINOR@.@CLIENT_VERSION_REVISION@.@CLIENT_VERSION_BUILD@, Copyright © 2009-@COPYRIGHT_YEAR@ @COPYRIGHT_HOLDERS_FINAL@</string>
```
Comment (fanquake) [inline share/qt/Info.plist.in]: Removing the version from here will mean it doesn't show up in the `Version` portion of the app info. I don't think that's what we want to do.
versions
